integration-tests/suites/socat.go

```diff
@@ -34,9 +34,17 @@ func (s *SocatTestSuite) SetupSuite() {
 
 	processImage := common.QaImage("quay.io/rhacs-eng/qa", "socat")
 
-	containerID, err := s.launchContainer("socat", "-e", "PORT1=80", "-e", "PORT2=8080", processImage, "/bin/sh", "-c", "socat TCP-LISTEN:$PORT1,fork STDOUT & socat TCP-LISTEN:$PORT2,fork STDOUT")
+	// the socat container only needs to exist long enough for use to run both
+	// socat commands. 300 seconds should be more than long enough.
+	containerID, err := s.launchContainer("socat", processImage, "/bin/sh", "-c", "/bin/sleep 300")
+	s.Require().NoError(err)
+
+	_, err = s.execContainer("socat", []string{"/bin/sh", "-c", "socat TCP-LISTEN:80,fork STDOUT &"})
+	s.Require().NoError(err)
 
+	_, err = s.execContainer("socat", []string{"/bin/sh", "-c", "socat TCP-LISTEN:8080,fork STDOUT &"})
 	s.Require().NoError(err)
+
```

Comment:
I think that this changes what this test checks for. Refer to https://github.com/stackrox/collector/pull/898 and https://issues.redhat.com/browse/ROX-13507. Also take a look at lines 44 and 45 in qa-tests-backend/src/test/groovy/ProcessesListeningOnPortsTest.groovy here https://github.com/stackrox/stackrox/pull/3546/files

That test was failing due to errors with running two instances of socat in the same command line. This test was a way of catching that error within collector and making it easier to solve. I named this test poorly. It should probably have been named something like "OpeningTwoPlopsInTheSameCommand".

Reply:
Ok thanks for the heads up, I'll rework this to better match the intended test case

Reply:
I am very confused about what running two processes from the same 'command line' actually tests. From the kernel's perspective and from Collector's they are just two processes, so that suggests that it might be the timing of those processes executing and possibly related to the scrape interval? I'm afraid those links do not shine much light on the overall context of this test.

In the future could we _please_ ensure that tests are well named and well documented with a much context as possible about what they're trying to test and any assumptions they are making.

Reply:
There is more information here https://srox.slack.com/archives/C02PQ7YUNSV/p1666921527207769

Basically the error was that in collector the two endpoints were incorrectly associated with the same PID, while process messages in collector had two different PIDs, which is correct.

I will also add more comments to the test to explain the error that it tests for.

Reply:
As discussed offline - I've tested this new approach against `3.12.x-3-g1cf8af5c3f` from [1cf8af5c](1cf8af5c3f06f625e6f50520f6e0b427d8676db3) and `3.12.x-10-g74e322fd28` from [74e322fd](https://github.com/stackrox/collector/commit/74e322fd) (the commit just prior to the fix being merged) as well as a new build from master. The test fails correctly on the two older versions, and passes on the new version. I think this gives us a decent amount of confidence that the test case is still covered by this approach.